swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/services/BillableUsageService.java

```diff
@@ -93,7 +101,7 @@ public BillableUsage produceMonthlyBillable(BillableUsage usage)
     Quantity<BillingUnit> contractAmount =
         Quantity.fromContractCoverage(usage, contractCoverage.getTotal());
     double applicableUsage =
-        Quantity.of(usage.getCurrentTotal())
+        Quantity.of(usage.getCurrentTotal() != null ? usage.getCurrentTotal() : 0.0)
```

Comment:
I see that the currentTotal is nullable, so this avoids a null pointer because the "of" method accepts the primitive double. 
Instead, we could change this "of" method to accept the Double class and use 0.0 if it's null. 
I think this change would make the of method more robust and improve the readibility of this logic.